assets/pages/monthly-contributions/components/paymentMethods.jsx

```diff
@@ -15,22 +16,29 @@ type PropTypes = {
   email: string,
   firstName: string,
   lastName: string,
+  error: ?string,
 };
 
 
 // ----- Component ----- //
 
 function PaymentMethods(props: PropTypes) {
 
-  let content = 'Please fill in all the fields above.';
+  let errorMessage = '';
+  let stripeButton = <StripePopUpButton email={props.email} callback={postCheckout} />;
+  if ((props.firstName === '' || props.lastName === '') && props.error === null) {
+    errorMessage = <ErrorMessage message={'Please fill in all the fields above.'} />;
+    stripeButton = '';
+  }
 
-  if (props.firstName !== '' && props.lastName !== '') {
-    content = <StripePopUpButton email={props.email} callback={postCheckout} />;
+  if (props.error !== null) {
+    errorMessage = <ErrorMessage message={'There was an error processing your payment. Please\u00a0try\u00a0again\u00a0later.'} />;
```

Comment:
What do we think could be causing a 500? Do we definitely want them to try again later (i.e. leave the page), or do we want to say `Please try again.`, indicating that they could do that right away with the reasonable expectation that it might work this time?